doc/design/architecture.md

```diff
@@ -82,10 +86,13 @@ Replacing --> Deleting
 
 ## Catalog Operator
 
-The Catalog Operator is responsible for resolving ClusterServiceVersions and the required resources they specify. It is also responsible for watching catalog sources for updates to packages in channels, and upgrading them (optionally automatically) to the latest available versions.
-A user that wishes to track a package in a channel creates a Subscription resource configuring the desired package, channel, and the catalog source from which to pull updates. When updates are found, an appropriate InstallPlan is written into the namespace on behalf of the user.
-Users can also create an InstallPlan resource directly, containing the names of the desired ClusterServiceVersions and an approval strategy and the Catalog Operator will create an execution plan for the creation of all of the required resources.
-Once approved, the Catalog Operator will create all of the resources in an InstallPlan; this should then independently satisfy the OLM Operator, which will proceed to install the ClusterServiceVersions.
+The Catalog Operator is responsible for resolving ClusterServiceVersions and the required resources they specify. It is also responsible for watching CatalogSources. It updates to packages in channels, and upgrading them (optionally automatically) to the latest available versions. It also starts and maintains pods required to provide the grpc endpoints from container images those CatalogSources identify.
```

Comment:
```
It updates to packages in channels, and upgrading them (optionally automatically) to the latest available versions.
```

I think you're missing a verb here? Maybe:

`It aggregates updates to packages in channels`?

Or something similar?

Reply:
I think splitting up the original sentence into 2 sentences may cost some confusion here. Perhaps, just keep the original sentence as it is.

Reply:
This is where I originally found confusion. I'll try again when I get a moment as my primary objective was to make it clear. :-) Thanks!

Reply:
I would suggest this:
`The Catalog Operator is responsible for resolving ClusterServiceVersions and the required resources they specify. It is also responsible for watching catalog sources for packages' updates in channels, and then upgrading those packages (optionally automatically) to the latest available versions.`
Does that sound better?

Reply:
When we're inside the cluster and have read the catalog source. What actually is changing when we "upgrade a package"?

I know an InstallPlan is triggered and the Subscription's status is updated. Of course the InstallPlan triggers some operand changes as required. The PackageManifest isn't related to the Catalog Operator. I think of the package as something that exists on the other side of the CatalogSource and not changed by OLM as a whole. If I want to change any part of the package, I replace the catalog, not a component of OLM.

Reply:
The package in this context is operator bundle AKA manifests yaml that make up the operator such as CSV, CRD and etc. Those manifests are pulled down inside the registry pod that is running.